src/schemas/user.schema.ts

```diff
@@ -14,10 +14,25 @@ export const patchUserSchema = z.object({
 	}),
 });
 
+/**
+ * This schema is used to validate `PATCH /users/marketing-emails/unsubscribe/:id` request
+ *
+ * @author arbaz23
+ */
+ export const updateUnsubscribeUserSchema = z.object({
+	params: z.object({
+		id: z.string({ required_error: "Id is required" }),
+	}),
+	body: z.object({
+		receiveMarketingEmails : z.boolean().optional(),
```

Comment:
This route is purely for unsubscribing, so you don't need to take `receiveMarketingEmails` as input